test_utils_helpers.py

```
@@ -0,0 +1,518 @@
+        assert result == expected
+    
+    def test_custom_indent(self):
+        """Test serializing with custom indentation"""
+        data = {"nested": {"key": "value"}}
+        result = safe_json_dumps(data, indent=4)
+        assert result.count(" ") > result.count("\n")  # More spaces due to indent=4
+    
+    def test_complex_object_with_datetime(self):
+        """Test serializing complex object with datetime (uses default=str)"""
+        from datetime import datetime
+        data = {"timestamp": datetime.now(), "value": 42}
```

Ruff (0.11.9)

`datetime.datetime.now()` called without a `tz` argument

(DTZ005)